docs: update flutter quickstart (#4652)
zitadel/zitadel: docs/docs/examples/login/flutter.md

https://github.com/zitadel/zitadel_flutter/blob/main/lib/main.dart#L14-L28
```

Second, the `MyHomePage` class will remain a stateful widget with
its title, we don't change any code here.

```dart reference
https://github.com/zitadel/zitadel_flutter/blob/main/lib/main.dart#L30-L37
```

What we'll change now, is the `_MyHomePageState` class to enable
authentication via ZITADEL and remove the counter button of the starter application. We'll show the username of the authenticated user.

We define the needed elements for our state:

```dart
var _busy = false;
var _authenticated = false;
var _username = '';
final storage = const FlutterSecureStorage();
```

Then the builder method, which does show the login button if you're not
authenticated, a loading bar if the login process is going on and
your name if you are authenticated:

```dart reference
https://github.com/zitadel/zitadel_flutter/blob/main/lib/main.dart#L119-L159
```

And finally the `_authenticate` method which calls the authorization endpoint,
then fetches the user info and stores the tokens into the secure storage.

```dart reference
https://github.com/zitadel/zitadel_flutter/blob/main/lib/main.dart#L45-L117
```

> Note that we have to use our http redirect URL for webapplications or otherwise use our custom scheme for Android and IOS devices.

Reviewer: ```suggestion
Note that we have to use our http redirect URL for web applications or otherwise use our custom scheme for Android and iOS devices.
```

To setup other platforms, read the documentation of the [Flutter Web Auth](https://pub.dev/packages/flutter_web_auth_2).

To ensure our application catches the callback URL, you have to create a `auth.html` file in the `/web`
folder with the following content:

```html reference
https://github.com/zitadel/zitadel_flutter/blob/main/web/auth.html
```

> Now, you can run your application for IOS and Android devices with

Reviewer: ```suggestion
Now, you can run your application for iOS and Android devices with
```

```bash
flutter run
```

or by directly selecting your device

```bash
flutter run -d iphone
```

> for Web make sure you run the application on your fixed port such that it maches your redirect URI in your ZITADEL application. We used 4444 as port before so the command would look like this:

Reviewer: ```suggestion
For Web make sure you run the application on your fixed port such that it matches your redirect URI in your ZITADEL application. We used 4444 as port before so the command would look like this:
```

```bash
flutter run -d chrome --web-port=4444
```

> Our Android and IOS Application opens ZITADEL's login within a custom tab, on Web a new tab is opened.

Reviewer: ```suggestion
Our Android and iOS Application opens ZITADEL's login within a custom tab, on Web a new tab is opened.
```
